src/main/java/org/checkerframework/specimin/JavaParserUtil.java

```diff
@@ -373,4 +360,101 @@ static String removeMethodReturnTypeAndAnnotationsImpl(String declAsString) {
     String result = methodWithoutReturnType.replace("< ", "<");
     return result;
   }
+
+  /**
+   * Checks to see if an expression (usually a field or method) is static.
+   *
+   * @param expr The expression
+   * @return Whether it is static or not
```

Comment:
Is this method really algorithmic, in the sense that it always gets the right answer? It seems to me that it is based on heuristics, and if that's a fair description it should be in the Javadoc.